src/compas_timber/connections/butt_joint.py

```diff
@@ -0,0 +1,231 @@
+        angles = face_dict.values()
+        angles, face_indices = zip(*sorted(zip(angles, face_indices)))
+        return self.cross_beam.faces[face_indices[1]], self.cross_beam.faces[face_indices[2]]
+
+    def front_back_surface_main(self):
+        face_dict = Joint._beam_side_incidence(self.cross_beam, self.main_beam, ignore_ends=True)
+        face_indices = face_dict.keys()
+        angles = face_dict.values()
+        angles, face_indices = zip(*sorted(zip(angles, face_indices)))
+        return self.main_beam.faces[face_indices[0]], self.main_beam.faces[face_indices[3]]
+
+    def back_surface_main(self):
+        face_dict = Joint._beam_side_incidence(self.main_beam, self.cross_beam, ignore_ends=True)
+        face_dict.sort(lambda x: x.values())
+        return face_dict.values()[3]
+
+    def subtraction_volume(self):
+        dir_pts = []
+        vertices = []
+        sides = self.side_surfaces_cross()
+        for i, side in enumerate(sides):
+            points = []
+            top_frame, bottom_frame = self.get_main_cutting_plane()
+            for frame in [top_frame, bottom_frame]:
+                for fr in self.front_back_surface_main():
+                    points.append(
+                        intersection_plane_plane_plane(
+                            Plane.from_frame(side), Plane.from_frame(frame), Plane.from_frame(fr)
+                        )
+                    )
+            pv = [subtract_vectors(pt, self.cross_beam.centerline.start) for pt in points]
+            dots = [dot_vectors(v, self.cross_beam.centerline.direction) for v in pv]
+            dots, points = zip(*sorted(zip(dots, points)))
+            min_pt, max_pt = points[0], points[-1]
+            top_line = Line(*intersection_plane_plane(Plane.from_frame(side), Plane.from_frame(top_frame)))
+            bottom_line = Line(*intersection_plane_plane(Plane.from_frame(side), Plane.from_frame(bottom_frame)))
+            top_min = Point(*closest_point_on_line(min_pt, top_line))
+            dir_pts.append(top_min)
+
+            top_max = Point(*closest_point_on_line(max_pt, top_line))
+            bottom_min = Point(*closest_point_on_line(min_pt, bottom_line))
+            bottom_max = Point(*closest_point_on_line(max_pt, bottom_line))
+            vertices.extend([Point(*top_min), Point(*top_max), Point(*bottom_max), Point(*bottom_min)])
+
+        center = (vertices[0] + vertices[1] + vertices[2] + vertices[3]) * 0.25
+
+        angle = angle_vectors_signed(
+            subtract_vectors(vertices[0], center), subtract_vectors(vertices[1], center), sides[0].zaxis
+        )
+        if angle > 0:
+            ph = Polyhedron(
+                vertices, [[0, 1, 2, 3], [1, 0, 4, 5], [2, 1, 5, 6], [3, 2, 6, 7], [0, 3, 7, 4], [7, 6, 5, 4]]
+            )
+        else:
+            ph = Polyhedron(
+                vertices, [[3, 2, 1, 0], [5, 4, 0, 1], [6, 5, 1, 2], [7, 6, 2, 3], [4, 7, 3, 0], [4, 5, 6, 7]]
+            )
+
+        return ph
```

Comment:
A couple of thoughts on this:

1. this is to me a big and mysterious chunk of code. Anyone potentially making changes to it in the future could benefit from the logic of it being more clearly defined either by: breaking it down into smaller chunks (e.g. using functions) or adding some comment in the beginning explaining the steps it takes towards calculating the volume.

2. On the strategic level, using meshes or breps to define a milling volume somewhat misses the point I think. Especially since we do have the parametric information used to create them. This logic will have to basically be duplicated (though in Beam frame) in the BTLx generator. Maybe we can re-think `MillVolume` and find a way to define it here in such way that will allow us to derive negative Brep for visualization but also to (with little effort) derive tool paths that can be later used for machining.